dApp/user/src/lib/Auth/AuthProvider.res

```diff
@@ -0,0 +1,41 @@
+type authContextType = {
+  isAuthorized: bool,
+  setIsAuthorized: ((bool => bool) => unit)
+}
+
+module AuthContext = {
+  let context = React.createContext({isAuthorized: false, setIsAuthorized: (_)=> () })
+
+  module Provider = {
+    let provider = React.Context.provider(context)
+
+    @react.component
+    let make = (~value, ~children) => {
+      React.createElement(provider, {"value": value, "children": children})
+    }
+  }
+}
+
+let getUserAuthStatus = (user) => Apollo.getAuthHeaders(~user)->Option.isSome
+
+@react.component
+let make = (~children) => {
+    let user = RootProvider.useCurrentUser()
+
+    let (isAuthorized, setIsAuthorized) = React.useState(_ => false)
+
+    React.useEffect1(() => {
+      let _ = setIsAuthorized(_ => user->getUserAuthStatus)
```

Comment:
The `getUserAuthStatus` takes in a `option<string>` right? Maybe we should make it take in a string?

```suggestion
	switch user {
	|Some(_) =>
      let isAuthorized = user->getUserAuthStatus->Option.map(usersAuth => {
        let _ = setAuthStatus("CurrentUserAuth", usersAuth)
      })->Option.isSome
      let _ = setIsAuthorized(_ => isAuthorized)
    | None =>
      let _ = setIsAuthorized(_ => Some("CurrentUserAuth"->getUserAuthStatus)
```

Reply:
Didn't test this code, probably got some errors, this is just the general idea.

Anyway, lets make this a new task :)

Reply:
#35

Reply:
> The `getUserAuthStatus` takes in a `option<string>` right? Maybe we should make it take in a string?

Takes in an `option<string>` as I'm using it also in the case that the user isn't connected to metamask. Probably better to refactor that logic out though, especially if we're moving away from having metamask and the DB auth necessarily connected.

Reply:
You can make it take in a string and use `let optionAuthStatus = optUser->Option.map(getUserAuthStatus)` which is cleaner IMO.